File: spec/python/test_expr_io_pos.py
import unittest

from expr_io_pos import ExprIoPos

class TestExprIoPos(unittest.TestCase):
    def test_expr_io_pos(self):
        r = ExprIoPos.from_file("src/expr_io_pos.bin")

        self.assertEqual(r.substream1.my_str, "CURIOSITY")
        self.assertEqual(r.substream1.body, bytearray([0x11, 0x22, 0x33, 0x44]))
        self.assertEqual(r.substream1.number, 0x42)

        self.assertEqual(r.substream2.my_str, "KILLED")
        self.assertEqual(r.substream2.body, b"a cat")
        self.assertEqual(r.substream2.number, 0x67)

File: spec/lua/test_expr_io_pos.lua
local luaunit = require("luaunit")

require("expr_io_pos")

TestExprIoPos = {}

function TestExprIoPos:test_expr_io_pos()
    local r = ExprIoPos:from_file("src/expr_io_pos.bin")

    luaunit.assertEquals(r.substream1.my_str, "CURIOSITY")
    luaunit.assertEquals(r.substream1.body, "\x11\x22\x33\x44")
    luaunit.assertEquals(r.substream1.number, 0x42)

    luaunit.assertEquals(r.substream2.my_str, "KILLED")
    luaunit.assertEquals(r.substream2.body, "a cat")
    luaunit.assertEquals(r.substream2.number, 0x67)
end
